packages/puppeteer-core/src/common/ElementHandle.ts

```diff
@@ -412,6 +421,61 @@ export class ElementHandle<
     return this.waitForSelector(`xpath/${xpath}`, options);
   }
 
+  /**
+   * Converts the current handle to the given element type.
+   *
+   * @example
+   *
+   * ```ts
+   * const element: ElementHandle<Element> = await page.$(
+   *   '.class-name-of-anchor'
+   * );
+   * // DO NOT DISPOSE `element`, this will be always be the same handle.
+   * const anchor: ElementHandle<HTMLAnchorElement> = await element.toElement(
+   *   'a'
+   * );
+   * ```
+   *
+   * @example
+   * Garbage handling is automatically handled on error, but you can turn this off.
+   *
+   * ```ts
+   * const element: ElementHandle<Element> = await page.$(
+   *   '.class-name-of-anchor'
+   * );
+   * try {
+   *   const anchor = await element.toElement('svg', {dispose: false});
+   * } catch (err) {
+   *   // You need to handle the trash since you specified `dispose: false`
+   *   element.dispose();
+   *   throw new err();
+   * }
+   * ```
+   *
+   * @param tagName - The tag name of the desired element type.
+   * @param options - Configures the behavior of this method.
+   * @throws An error if the handle does not match. **This will dispose the
+   * current handle.** You can disable this behavior in the options.
+   */
+  async toElement<
+    K extends keyof HTMLElementTagNameMap | keyof SVGElementTagNameMap
+  >(
+    tagName: K,
+    options: ToElementOptions = {}
+  ): Promise<HandleFor<ElementFor<K>>> {
+    const {dispose = true} = options;
+    const isMatchingTagName = await this.evaluate((node, tagName) => {
+      return node.nodeName === tagName.toUpperCase();
+    }, tagName);
+    if (!isMatchingTagName) {
+      if (dispose) {
+        await this.dispose();
```

Comment:
I would prefer the method to have a single responsibility: let's return null if the check fail and allow the client to dispose or not-dispose outside of this method. Alternatively, we can have a second method that is more strict and indicates in its name that the handle might be disposed (possible names `toElementOrDispose`).

Reply:
would it work for custom elements?

Reply:
Yes. It also works with `class Test extends HTMLDivElement` because these type of custom elements still use `DIV` as their node name.

Reply:
> let's return null if the check fail and allow the client to dispose or not-dispose outside of this method.

The reason I want to throw is because there is an expectation that the element *will* be the correct. We could write `toElementOrThrow` though. For disposal, I would still like to keep the `dispose` option.